OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/gpx/GpxTrackAnalysis.kt

```diff
@@ -446,8 +450,11 @@ class GpxTrackAnalysis {
 			}
 			processElevationDiff(s)
 		}
-
+		totalDistanceWithoutGaps = _totalDistanceWithoutGaps
 		totalDistance = _totalDistance
+		if(splitSegments.size == 1 && !splitSegments[0].segment.isGeneralSegment()) {
+			totalDistanceWithoutGaps = totalDistance
+		}
```

Comment:
It looks strange. Is it needed because there are different values in totalDistance and totalDistanceWithoutGaps when we do not have multiple segments?

Reply:
yes. in case when we have only one segment we don't have "general segment" and totalDistanceWithoutGaps is not calculated, but it can be referenced in code